Platform support (#152)
vectorgrp/sil-kit: README.rst

    cmake -D SILKIT_BUILD_DOCS=ON -B _build
    cmake --build _build --target Doxygen

Platform support
----------------------------------------

SIL Kit provides three tiers of platform support

.. list-table:: Support Tiers

   * - Tier 1
     - Essential targets. Continuously tested and
       official binary packages are provided.

>    * - Tier 1

Reviewer: ```suggestion
   * - Tier 2
```

     - Officially supported targets. Continuously tested 
       but NO binary packages are provided
   * - Tier 3
     - | Targets for which we have (limited) build support but which
       | are not continuously tested and no packages are provided from us

> A platform
> hereby is defined by a combination of the used operating system (OS), the CPU
> Architecture (eg. x86 or ARM64) and the compiler/toolchain used. An example
> would be `Ubuntu 20.04 x86_64 Clang 10`.

Reviewer: ```suggestion
A platform is hereby defined by the combination of the used operating system (OS), the CPU architecture (eg. x86 or ARM64) and the compiler/toolchain used. 
For example,  `Ubuntu 20.04 x86_64 Clang 10`.
```

SIL Kit should compile and run on

> any POSIX platform. If you have feedback for different targets or platforms
> please report via the GitHub issues, thanks!

Reviewer: ```suggestion
any POSIX platform. 
If you have feedback for targets or platforms not listed here, please report via the GitHub issues, thanks!
```

A target may be upgraded to
`Tier 1` once we have continuous testing for it in place and we have binary
packages available for it.

Tier 1
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Essential targets. Automatically tested and provided as official binary packages.

.. list-table:: Platform Support
   :header-rows: 1

   * - OS
     - Architecture
     - Notes
   * - Windows
     - 64bit (x86_64)
     - MSVC 19 with Toolset 14.1
   * - Windows
     - 32bit (x86)
     - MSVC 19 with Toolset 14.1
   * - Ubuntu 18.04
     - amd64
     - GCC 8
   * - Ubuntu 20.04
     - amd64

>      - Clang 10

Reviewer: ```suggestion
     - Clang 10, `.deb`
```

Tier 2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Officially supported and automatically tested. NO binary packages provided

.. list-table:: Platform Support
   :header-rows: 1

   * - Windows
     - 64bit (x86_64)

>      - MSYS2/Mingw: GCC 14

Reviewer: it';s good to mention MSYS2 here, as there are many different MinGW builds out there. Not sure if our CI workflow files are enough as documentation on how to compile, or if we need a dedicated `Building SIL Kit on different architectures` documentation in the future.

Author: A section on MinGW would be good, since some of the problems with MingGW/MSYS2 are which toolchain to install and use, which is not too intuitive for newcomers to the MinGW platform.

   * - Ubuntu 22.04
     - amd64
     - | GCC 11/Clang 18
       | + Address Sanitizer
       | + Undefined Behaviour Sanitizer
       | + Thread Sanitizer
   * - Ubuntu 22.04
     - ARM64
     - Clang 18
   * - MAC OS
     - ARM64/M1
     - AppleClang 15

Tier 3
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Build and tested by individual contributors or users.
Since these are not part of the CI pipeline, compatibility with these platforms can be broken at any time!

.. list-table:: Platform Support
   :header-rows: 1

   * - Platform
     - Architecture
     - Notes
   * - Ubuntu 24.04
     - amd64
     - | GCC 13
       | Clang 18
   * - QNX 7.1 RTOS
     - X86 64bit

>      - QNX GCC 8

Reviewer: should we note that we are working on QNX CI integration? or is this the wrong place

Author: I would find a different place for this kind of NEWS. Maybe a Roadmap section in the docs somewhere?
